src/NHapi.Base/Parser/PipeParser.cs

```diff
@@ -24,885 +24,960 @@
+        /// This seems to be more convenient than checking to see if they are needed
+        /// while we are building the encoded string.
+        /// </summary>
+        private static string StripExtraDelimiters(string in_Renamed, char delim)
+        {
+            var chars = in_Renamed.ToCharArray();
+
+            // search from back end for first occurrence of non-delimiter ...
+            var c = chars.Length - 1;
+            var found = false;
+            while (c >= 0 && !found)
+            {
+                if (chars[c--] != delim)
+                {
+                    found = true;
+                }
+            }
+
+            var ret = string.Empty;
+            if (found)
+            {
+                ret = new string(chars, 0, c + 2);
+            }
+
+            return ret;
+        }
+
+        /// <returns> true if the segment is MSH, FHS, or BHS.  These need special treatment
+        /// because they define delimiters.
+        /// </returns>
+        /// <param name="theSegmentName">
+        /// </param>
+        private static bool IsDelimDefSegment(string theSegmentName)
+        {
+            var is_Renamed = false;
+            if (theSegmentName.Equals("MSH") || theSegmentName.Equals("FHS") || theSegmentName.Equals("BHS"))
+            {
+                is_Renamed = true;
+            }
+
+            return is_Renamed;
+        }
+
+        /// <summary> Fills a field with values from an unparsed string representing the field.  </summary>
+        /// <param name="destinationField">the field Type.
+        /// </param>
+        /// <param name="data">the field string (including all components and subcomponents; not including field delimiters).
+        /// </param>
+        /// <param name="encodingCharacters">the encoding characters used in the message.
+        /// </param>
+        private static void Parse(IType destinationField, string data, EncodingCharacters encodingCharacters)
+        {
+            var components = Split(data, Convert.ToString(encodingCharacters.ComponentSeparator));
+            for (var i = 0; i < components.Length; i++)
+            {
+                var subcomponents = Split(components[i], Convert.ToString(encodingCharacters.SubcomponentSeparator));
+                for (var j = 0; j < subcomponents.Length; j++)
+                {
+                    var val = subcomponents[j];
+                    if (val != null)
+                    {
+                        val = Escape.UnescapeText(val, encodingCharacters);
+                    }
+
+                    Terser.GetPrimitive(destinationField, i + 1, j + 1).Value = val;
+                }
+            }
+        }
+
+        /// <summary>Returns the component or subcomponent separator from the given encoding characters. </summary>
+        private static char GetSeparator(bool subComponents, EncodingCharacters encodingChars)
+        {
+            char separator;

```

Comment:
```csharp
return subComponents 
    ? encodingChars.SubcomponentSeparator 
    : encodingChars.ComponentSeparator;
```

?

Reply:
this method isnt even used, so have removed it